UnityProject/Assets/Scripts/Items/IDCard.cs

```diff
@@ -137,68 +126,46 @@ public void OnInventoryMoveServer(InventoryMove info)
 	/// <param name="jobType">job on the card</param>
 	/// <param name="allowedAccess">what the card can access</param>
 	/// <param name="name">name listed on card</param>
-	private void Initialize(IDCardType idCardType, JobType jobType, List<Access> allowedAccess, string name)
+	private void Initialize(IDCardType idCardType, JobType newJobType, List<Access> allowedAccess, string name)
 	{
 		//Set all the synced properties for the card
 		SyncName(registeredName, name);
-		this.jobType = jobType;
+		SyncJobType(jobType, newJobType);
 		SyncIDCardType(idCardType, idCardType);
 		ServerAddAccess(allowedAccess);
 	}
 
-	private void InitCard()
+	public void SyncName(string oldName, string newName)
 	{
-		if (isInit)
-		{
-			return;
-		}
-
-		isInit = true;
-		accessSyncList.Callback += SyncAccess;
-
-		//This will add the access from ManuallyAddedAccess list
-		if (isServer)
-		{
-			if (manuallyAddedAccess.Count > 0)
-			{
-				ServerAddAccess(manuallyAddedAccess);
-				idCardType = manuallyAssignCardType;
-			}
-		}
-	}
-
-	//Sync all of the current in game ID's throughout the map with new players
-	private IEnumerator WaitForLoad()
-	{
-		yield return WaitFor.Seconds(3f);
-		SyncName(registeredName, registeredName);
-		SyncJobType(jobType, jobType);
-		SyncIDCardType(idCardType, idCardType);
-	}
-
-	public void SyncAccess(SyncList<int>.Operation op, int index, int oldItem, int newItem)
-	{
-		//Do anything special when the synclist changes on the client
+		registeredName = newName;
+		RenameIDObject();
 	}
 
-	public void SyncName(string oldName, string name)
+	public void SyncJobType(JobType oldJobType, JobType newJobType)
 	{
-		EnsureInit();
-		registeredName = name;
+		jobType = newJobType;
+		RenameIDObject();
 	}
 
-	public void SyncJobType(JobType oldJobType, JobType jobType)
+	private void RenameIDObject()
 	{
-		EnsureInit();
-		this.jobType = jobType;
+		var newName = "";
+		if (!RegisteredName.IsNullOrEmpty())
+		{
+			newName += $"{RegisteredName}'s ";
+		}
+		newName += "ID Card";
+		if (JobType != JobType.NULL)
+		{
+			newName += $" ({JobType})";
+		}
+		itemAttributes.ServerSetArticleName(newName);
 	}
 
 	public void SyncIDCardType(IDCardType oldCardType, IDCardType cardType)
 	{
-		EnsureInit();
 		idCardType = cardType;
-		IDCardType cType = IDCardType;
-		switch (cType)
+		switch (idCardType)
```

Comment:
add the default case for safety. You can log an error and use a the standard sprite if the code ever got that point.